ethereum/contracts/ShutdownSwitch.sol

```diff
@@ -179,11 +174,54 @@ abstract contract ShutdownSwitch {
     function computeNumVotesShutdown(Structs.GuardianSet memory gs) private view returns (uint16) {
         uint16 votesToShutdown = 0;
         for (uint idx = 0; (idx < gs.keys.length); idx++) {
-            if (shutdownVotes[gs.keys[idx]]) {
+            if (_getShutdownVote(gs.keys[idx])) {
                 votesToShutdown++;
             }
         }
 
         return votesToShutdown;
     }
+    
+    /// @dev Since other contracts will inherit from this one, it cannot have any of its own local state.
+    /// This is because that would affect the layout of local state in the inheriting contracts. To prevent this,
+    /// we use the storage slot library to put our state in deterministic storage.
```

Comment:
we're not using the storage slot library here, just doing the inline assembly directly